platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ChatterNetwork.java

```diff
@@ -0,0 +1,382 @@
+import com.swirlds.platform.metrics.ReconnectMetrics;
+import com.swirlds.platform.metrics.SyncMetrics;
+import com.swirlds.platform.network.ConnectionTracker;
+import com.swirlds.platform.network.NetworkMetrics;
+import com.swirlds.platform.network.communication.NegotiationProtocols;
+import com.swirlds.platform.network.communication.NegotiatorThread;
+import com.swirlds.platform.network.communication.handshake.VersionCompareHandshake;
+import com.swirlds.platform.network.topology.NetworkTopology;
+import com.swirlds.platform.network.topology.StaticConnectionManagers;
+import com.swirlds.platform.observers.EventObserverDispatcher;
+import com.swirlds.platform.reconnect.DefaultSignedStateValidator;
+import com.swirlds.platform.reconnect.FallenBehindManagerImpl;
+import com.swirlds.platform.reconnect.ReconnectController;
+import com.swirlds.platform.reconnect.ReconnectProtocol;
+import com.swirlds.platform.reconnect.ReconnectThrottle;
+import com.swirlds.platform.reconnect.emergency.EmergencyReconnectProtocol;
+import com.swirlds.platform.state.EmergencyRecoveryManager;
+import com.swirlds.platform.state.SwirldStateManager;
+import com.swirlds.platform.state.signed.SignedState;
+import com.swirlds.platform.sync.ShadowGraph;
+import com.swirlds.platform.sync.ShadowGraphSynchronizer;
+import com.swirlds.platform.threading.PauseAndClear;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Supplier;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * Encapsulates a chatter-based gossip network.
+ */
+public class ChatterNetwork extends GossipNetwork implements Startable {
+
+    private final ChatterCore<GossipEvent> chatterCore;
+    private final SyncMetrics syncMetrics;
+    private final EventTaskCreator eventTaskCreator;
+    private final EmergencyRecoveryManager emergencyRecoveryManager;
+    private final ShadowGraph shadowGraph;
+    private final ReconnectController reconnectController;
+    private final ReconnectThrottle reconnectThrottle;
+    private final Supplier<Consensus> consensusSupplier;
+    private final NotificationEngine notificationEngine;
+    private final StateManagementComponent stateManagementComponent;
+    private final FallenBehindManagerImpl fallenBehindManager;
+    private final boolean startedFromGenesis;
+    private final ReconnectMetrics reconnectMetrics;
+    private final ChatterEventMapper chatterEventMapper = new ChatterEventMapper();
+    private final QueueThread<EventIntakeTask> intakeQueue;
+    private final SequenceCycle<EventIntakeTask> intakeCycle;
+    private final EventCreatorThread eventCreatorThread;
+    private final ChatterEventCreator chatterEventCreator;
+    private final EventLinker eventLinker;
+
+    /**
+     * A list of threads that execute the chatter protocol.
+     */
+    private final List<StoppableThread> chatterThreads = new LinkedList<>();
+
+    public ChatterNetwork(
```

Comment:
I'll add a short description, but I'm specifically NOT going to write javadocs for ~30 parameters. 🙃  This class is a huge mess and a temporary solution. We should tolerate this class only because it's a lot better than the previous iteration.